src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ContentCacheGenerationService.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Publisher.Services.Interfaces;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+
+namespace GovUk.Education.ExploreEducationStatistics.Publisher.Services
+{
+    public class ContentCacheGenerationService : IContentCacheGenerationService
+    {
+        private readonly IContentService _contentService;
+        private readonly IReleaseService _releaseService;
+        private readonly IPublicationService _publicationService;
+        private readonly IDownloadService _downloadService;
+        private readonly IMethodologyService _methodologyService;
+        private readonly CloudBlobContainer _cloudBlobContainer;
+
+        private const string ContainerName = "cache";
+
+        public ContentCacheGenerationService(IDownloadService downloadService, IMethodologyService methodologyService,
+            IContentService contentService, IReleaseService releaseService, IPublicationService publicationService,
+            IConfiguration config)
+        {
+            _contentService = contentService;
+            _releaseService = releaseService;
+            _publicationService = publicationService;
+            _downloadService = downloadService;
+            _methodologyService = methodologyService;
+
+            _cloudBlobContainer = GetCloudBlobContainer(config.GetConnectionString("PublicStorage"));
+        }
+
+        public async Task<bool> CleanAndRebuildFullCache()
+        {
+            try
+            {
+                var trees = await UpdateTrees();
+
+                var publications = await UpdatePublicationsAndReleases();
+
+                var methodologies = await UpdateMethodologies();
+
+                return trees && methodologies && publications;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
```

Comment:
Not sure whether this will go to the function logger or not - should probs use the ILogger stuff anyway.

Reply:
BTW - There some real nastiness with the logging output i.e. it doesn't output unless you have the correct configuration in the host.json file (see Processor function)

Reply:
Agree, sort out the logging of services by following `Data.Processor/host.json` as an example.